workflow/scores/_circles.py

```diff
@@ -45,14 +45,20 @@ def project_circle(circle, image=None, shape=None,
     if negative:
         value = - value
 
+    # only coords within the image
+    coords_T = np.array([[i, j] for (i, j) in zip(coords[0], coords[1])])
+    image_mask = (coords[0] >= 0) & (coords[0] < image.shape[0])
+    image_mask = image_mask & (coords[1] >= 0) & (coords[1] < image.shape[1])
+    coords = coords_T[image_mask].T
+    coords = (coords[0], coords[1])
```

Comment:
Unless I'm missing something here, this is unnecessary since the `shape` keyword or the `circle_coords` method above is here for that very reason.

Reply:
It crashed when I tried.

Reply:
Just committed a simple detector, the one that crashed with the previous version. Maybe it crashed because `shape` was not set properly when it was `None`. In any case, if you can fix it in a simpler way, don't hesitate, just make `submissions/full` pass.

Reply:
Not your fault indeed, in the `circle_map` function, the `shape` is not passed to the `project_circle` method.. could you correct that ?

Reply:
Done